src/application.rs

```diff
@@ -92,6 +93,11 @@ impl Application for RusticApp {
         // Configure components
         self.state.components_mut().after_config(&config)?;
 
+        // set all given environment variables
+        for (env, value) in config.global.env.iter() {
+            env::set_var(env, value);
+        }
+
```

Comment:
It only sets the var for the local process (and subprocesses, oft course). So, no need to clean up anything

Reply:
I also would not print a warning, this feature is actually just a feature which allows people to put a 'MYVAR=xxx rustic' call  into the config file instead of using a script but should else behave exactly identically...